### PR TITLE
feat: add framework name to the test classes BSP request

### PR DESCRIPTION
**Pull request by kpodsiad**
I've added `[skip ci]` because for now I only want to see what do you folks think about this PR.
This PR will stay as a draft until changes in `bsp4j` are merged, for now I've used locally published `bsp4j` version.

It is an implementation of the BSP proposal which is described [here](https://github.com/build-server-protocol/build-server-protocol/issues/296#issuecomment-1041525725). This proposal adds an optional framework field to the returned class.

There's already implementation for [bloop](https://github.com/scalacenter/bloop/pull/1695).

### CODE REVIEW

**Review by lefou — COMMENTED** [review]
Looks reasonable to me. Please update as appropriate and remove the draft status when you think it is mergeable. Thank you!

File: build.sc [hunk 1]
```diff
@@ -133,7 +133,7 @@ object Deps {
   val utest = ivy"com.lihaoyi::utest:0.7.11"
   val windowsAnsi = ivy"io.github.alexarchambault.windows-ansi:windows-ansi:0.0.3"
   val zinc = ivy"org.scala-sbt::zinc:1.6.1"
-  val bsp = ivy"ch.epfl.scala:bsp4j:2.0.0"
+  val bsp = ivy"ch.epfl.scala:bsp4j:2.0.0:+70-f6e47d42-SNAPSHOT"
```
Comment (lefou) [inline build.sc:136]: Version doesn't look right. 

```suggestion
  val bsp = ivy"ch.epfl.scala:bsp4j:2.0.0+70-f6e47d42-SNAPSHOT"
```

**Review by lefou — COMMENTED** [review]
Surprisingly, this is only for Scala projects and not for e.g. Java projects. Anyhow, I'm happy to merge, although this feature is probably not very relevant for Mill users, as we only support one test framework per module.

**Comment by kpodsiad** [comment]
Resolution is failing because `ch.epfl.scala:bsp4j:2.0.0+70-f6e47d42-SNAPSHOT` is being resolved on the sonatype releases, but it's a snapshot. I'll wait until the official release is published.

File: bsp/src/mill/bsp/MillBuildServer.scala [hunk 2]
```diff
@@ -625,6 +627,12 @@ class MillBuildServer(
       new CleanCacheResult(msg, cleaned)
     }
 
+  override def debugSessionStart(debugParams: DebugSessionParams)
+      : CompletableFuture[DebugSessionAddress] =
+    completable(s"debugSessionStart ${debugParams}") { state =>
+      throw new NotImplementedError("debugSessionStart endpoint is not implemented")
+    }
```
Comment (kpodsiad) [inline bsp/src/mill/bsp/MillBuildServer.scala:634]: There's a `DebugProvider` capability that is not set by the user so it's safe to throw an error here from the BSP client perspective.

**Comment by lefou** [comment]
> @lefou I updated PR with the newly released bsp4j version.
> 
> > Surprisingly, this is only for Scala projects and not for e.g. Java projects.
> 
> Do you mean that method names of method and classes contain `ScalaTestClasses`?

I mean, the framework name is only relevant for Scala projects, as it is implemented in `ScalaBuildServer`. So pure Java projects, for which we also `sbt.testing`, don't profit from this test framework selection.

**Comment by kpodsiad** [comment]
@lefou It's true that it's more general than Scala and there was a small discussion about it [here](https://github.com/build-server-protocol/build-server-protocol/pull/304#pullrequestreview-941147293).

**Comment by lefou** [comment]
Thanks for the pointer. I think, this would be the right move, too.

**Pull request merged**